x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/percentile.tsx

```diff
@@ -0,0 +1,190 @@
+    defaultMessage: 'Percentile',
+  }),
+  input: 'field',
+  getPossibleOperationForField: ({ aggregationRestrictions, aggregatable, type: fieldType }) => {
+    if (fieldType === 'number' && aggregatable && !aggregationRestrictions) {
+      return {
+        dataType: 'number',
+        isBucketed: false,
+        scale: 'ratio',
+      };
+    }
+  },
+  isTransferable: (column, newIndexPattern) => {
+    const newField = newIndexPattern.getFieldByName(column.sourceField);
+
+    return Boolean(
+      newField &&
+        newField.type === 'number' &&
+        newField.aggregatable &&
+        !newField.aggregationRestrictions
+    );
+  },
+  getDefaultLabel: (column, indexPattern, columns) =>
+    ofName(indexPattern.getFieldByName(column.sourceField)!.displayName, column.params.percentile),
+  buildColumn: ({ field, previousColumn }) => {
+    const existingFormat =
+      previousColumn?.params && 'format' in previousColumn?.params
+        ? previousColumn?.params?.format
+        : undefined;
+    const existingPercentileParam =
+      previousColumn?.operationType === 'percentile' && previousColumn?.params.percentile;
+    const newPercentileParam = existingPercentileParam || DEFAULT_PERCENTILE_VALUE;
+    return {
+      label: ofName(field.displayName, newPercentileParam),
+      dataType: 'number',
+      operationType: 'percentile',
+      sourceField: field.name,
+      isBucketed: false,
+      scale: 'ratio',
+      params: {
+        format: existingFormat,
+        percentile: newPercentileParam,
+      },
+    };
+  },
+  onFieldChange: (oldColumn, field) => {
+    return {
+      ...oldColumn,
+      label: ofName(field.displayName, oldColumn.params.percentile),
+      sourceField: field.name,
+    };
+  },
+  toEsAggsFn: (column, columnId, _indexPattern) => {
+    return buildExpressionFunction<AggFunctionsMapping['aggPercentiles']>('aggPercentiles', {
+      id: columnId,
+      enabled: true,
+      schema: 'metric',
+      field: column.sourceField,
+      percents: [column.params.percentile],
+    }).toAst();
+  },
+  getEsAggsSuffix: (column) => {
+    const value = column.params.percentile;
+    return `.${value}`;
+  },
+  getErrorMessage: (layer, columnId, indexPattern) =>
+    getInvalidFieldMessage(layer.columns[columnId] as FieldBasedIndexPatternColumn, indexPattern),
+  paramEditor: function PercentileParamEditor({
+    layer,
+    updateLayer,
+    currentColumn,
+    columnId,
+    indexPattern,
+  }) {
+    const [inputValue, setInputValue] = useState(String(currentColumn.params.percentile));
+
+    const inputValueAsNumber = Number(inputValue);
+    // an input is value if it's not an empty string, parses to a valid number, is between 0 and 100 (inclusive)
+    // and is an integer
+    const inputValueIsValid =
+      inputValue !== '' &&
+      !Number.isNaN(inputValueAsNumber) &&
+      inputValueAsNumber >= 0 &&
+      inputValueAsNumber <= 100 &&
+      inputValueAsNumber % 1 === 0;
```

Comment:
Can you move some of the basic integer checking into helpers? We're using it in 3 places now.

Also, why is 0 a valid percentile?

Reply:
> Also, why is 0 a valid percentile?

Interesting, Elasticsearch treats the 0th percentile as min and 100th percentile as max, but according to wikipedia this is not how it works in general:

> Note that in theory the 0th percentile falls at negative infinity and the 100th percentile at positive infinity, although in 
many practical applications, such as test results, natural lower and/or upper limits are enforced.

I suggest we follow the common definition of percentile here and make the allowed range 1-99 exclusive, wdyt? I would also be fine with following Elasticsearch